File: backend/src/Workora.Application/Features/Assets/DTOs/AssetDto.cs
using Workora.Domain.Enums;
using Workora.Shared.Responses;

namespace Workora.Application.Features.Assets.DTOs;

/// <summary>
/// DTO representing an equipment asset.
/// </summary>
public record AssetDto(
    int Id,
    Guid Uuid,
    int CompanyId,
    string Name,
    string AssetTag,
    string? SerialNumber,
    string Category,
    AssetStatus Status,
    decimal? PurchaseCost,
    DateOnly? PurchaseDate,
    string? CurrentAssignedEmployeeName,
    int? CurrentAssignedEmployeeId,
    bool IsActive,
    DateTimeOffset CreatedAt);

File: frontend/workora-ui/src/app/data/dtos/asset.dto.ts
export interface AssetDto {
  id: number;
  uuid: string;
  companyId: number;
  name: string;
  assetTag: string;
  serialNumber?: string | null;
  category: string;
  status: string;
  purchaseCost?: number | null;
  purchaseDate?: string | null;
  currentAssignedEmployeeName?: string | null;
  currentAssignedEmployeeId?: number | null;
  isActive: boolean;
  createdAt: string;
}

export interface CreateAssetRequestDto {
  companyId: number;
  name: string;
  assetTag: string;
  category: string;
  serialNumber?: string | null;
  purchaseCost?: number | null;
  purchaseDate?: string | null;
}

export interface AssignAssetRequestDto {
  assetId: number;
  employeeId: number;
  assignedDate: string;
}

export interface ReturnAssetRequestDto {
  assetId: number;
  returnedDate: string;
  condition?: string | null;
}
